flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplit.java

```diff
@@ -57,38 +69,64 @@ public boolean equals(Object o) {
             return false;
         }
         HybridSourceSplit that = (HybridSourceSplit) o;
-        return sourceIndex == that.sourceIndex && wrappedSplit.equals(that.wrappedSplit);
+        return sourceIndex == that.sourceIndex
+                && Arrays.equals(wrappedSplitBytes, that.wrappedSplitBytes);
```

Comment:
Don't need the `splitId` equal?

Reply:
Not needed because `splitId` is already part of `wrappedSplitBytes`.